xds/src/main/java/io/grpc/xds/XdsClientMetricReporterImpl.java

```diff
@@ -143,7 +143,13 @@ void reportCallbackMetrics(BatchRecorder recorder, XdsClient xdsClient) {
       Map<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByType =
           getResourceMetadataCompleted.get(10, TimeUnit.SECONDS);
 
-      computeAndReportResourceCounts(metadataByType, callback);
+      ListenableFuture<Map<XdsResourceType<?>, Map<String, String>>>
+              getResourceAuthorityCompleted = xdsClient.getSubscribedResourcesAuthoritySnapshot();
```

Comment:
This fixes the data race, but there's still a regular race. The two snapshots may not contain all the same types/resources. We need to either pass the authorities through an existing flow or re-compute it from the resource name.